test/index.ts

```diff
@@ -270,9 +286,13 @@ function testDeepPartial() {
     Assert<IsExact<DeepPartial<undefined>, undefined>>,
     Assert<IsExact<DeepPartial<null>, null>>,
     Assert<IsExact<DeepPartial<Function>, Function>>,
+    Assert<IsExact<DeepPartial<ExtendedFunction>, Function & { args: 0 }>>,
     Assert<IsExact<DeepPartial<Date>, Date>>,
+    Assert<IsExact<DeepPartial<ExtendedDate>, Date & { today(): number }>>,
     Assert<IsExact<DeepPartial<Error>, Error>>,
+    Assert<IsExact<DeepPartial<ExtendedError>, Error & { code: string }>>,
     Assert<IsExact<DeepPartial<RegExp>, RegExp>>,
+    Assert<IsExact<DeepPartial<ExtendedRegExp>, RegExp & { caseSensitive: true }>>,
```

Comment:
At least we know that here we don't apply `Partial` (or don't encourage to do that)